index.js

```diff
@@ -310,28 +303,31 @@ async function handleSensorRequest(currentLocationId){
     let statemachine = new SessionState(currentLocationId);
     let currentState = await statemachine.getNextState(db, redis);
     let stateobject = await redis.getLatestLocationStatesData(currentLocationId)
-    let prevState = stateobject.state;
+    let prevState
+    if(!stateobject){
+        prevState=STATE.RESET
+    }else{
+        prevState = stateobject.state;
+    }
     let location = await db.getLocationData(currentLocationId);
 
     // Check the XeThru Heartbeat
-    await checkHeartbeat(currentLocationId);
-
+    if(!helpers.isTestEnvironment()){
+        await checkHeartbeat(currentLocationId);
+    }
     console.log(`${currentLocationId}: ${currentState}`);
 
     // Get current time to compare to the session's start time
 
     var location_start_time = start_times[currentLocationId]
+    console.log(`location_start_time: ${location_start_time}`)
     if(location_start_time != null && location_start_time != undefined){
-        var today = new Date();
-        var date = today.getFullYear()+'-'+(today.getMonth()+1)+'-'+today.getDate();
-        var time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
-        var dateTime_string = date+' '+time;
 
-        var dateTime = new Date(dateTime_string);
         var start_time_sesh = new Date(location_start_time);
+        var now = new Date()
 
         // Current Session duration so far:
-        var sessionDuration = (dateTime - start_time_sesh)/1000;
+        var sessionDuration = (now - start_time_sesh)/1000;
```

Comment:
Thanks for doing this - I hope it resolves the issue of negative time!

Reply:
it does!